pkg/fleetautoscalers/fleetautoscalers.go

```diff
@@ -41,94 +41,147 @@ var client = http.Client{
 
 // computeDesiredFleetSize computes the new desired size of the given fleet
 func computeDesiredFleetSize(fas *autoscalingv1.FleetAutoscaler, f *agonesv1.Fleet) (int32, bool, error) {
-
 	switch fas.Spec.Policy.Type {
 	case autoscalingv1.BufferPolicyType:
 		return applyBufferPolicy(fas.Spec.Policy.Buffer, f)
 	case autoscalingv1.WebhookPolicyType:
 		return applyWebhookPolicy(fas.Spec.Policy.Webhook, f)
 	}
 
-	return f.Status.Replicas, false, errors.New("wrong policy type, should be one of: Buffer, Webhook")
+	return 0, false, errors.New("wrong policy type, should be one of: Buffer, Webhook")
 }
 
-func applyWebhookPolicy(w *autoscalingv1.WebhookPolicy, f *agonesv1.Fleet) (int32, bool, error) {
-	faReq := autoscalingv1.FleetAutoscaleReview{
-		Request: &autoscalingv1.FleetAutoscaleRequest{
-			UID:       uuid.NewUUID(),
-			Name:      f.Name,
-			Namespace: f.Namespace,
-			Status:    f.Status,
-		},
-		Response: nil,
-	}
-	b, err := json.Marshal(faReq)
-	urlStr := ""
+func buildURLFromWebhookPolicy(w *autoscalingv1.WebhookPolicy) (*url.URL, error) {
+	var u *url.URL
+	var err error
+
 	if w.URL != nil {
-		urlStr = *w.URL
-	}
-	var faResp autoscalingv1.FleetAutoscaleReview
-	servicePath := ""
-	if w.Service != nil {
-		if w.Service.Path != nil {
-			servicePath = *w.Service.Path
+		if *w.URL == "" {
+			return nil, errors.New("URL was not provided")
 		}
+
+		u, err = url.ParseRequestURI(*w.URL)
 		if err != nil {
-			return f.Status.Replicas, false, err
+			return nil, err
+		}
+	} else {
+		if w.Service.Name == "" {
+			return nil, errors.New("service name was not provided")
+		}
+
+		var servicePath string
```

Comment:
I'm not clear on why service Path is stored as a local variable but service namespace and the ca bundle are set on the webhook policy.

Reply:
changed this logic